src/Compilers/CSharp/Portable/Operations/CSharpOperationFactory.cs

```diff
@@ -445,12 +455,35 @@ private IPropertyReferenceExpression CreateBoundAnonymousPropertyDeclarationOper
             return new LazyPropertyReferenceExpression(property, instance, member, argumentsInEvaluationOrder, _semanticModel, syntax, type, constantValue, isImplicit);
         }
 
-        private IObjectCreationExpression CreateBoundObjectCreationExpressionOperation(BoundObjectCreationExpression boundObjectCreationExpression)
+        private IOperation CreateBoundObjectCreationExpressionOperation(BoundObjectCreationExpression boundObjectCreationExpression)
         {
-            IMethodSymbol constructor = boundObjectCreationExpression.Constructor;
+            MethodSymbol constructor = boundObjectCreationExpression.Constructor;
+            SyntaxNode syntax = boundObjectCreationExpression.Syntax;
+            ITypeSymbol type = boundObjectCreationExpression.Type;
+            Optional<object> constantValue = ConvertToOptional(boundObjectCreationExpression.ConstantValue);
+            bool isImplicit = boundObjectCreationExpression.WasCompilerGenerated;
+            
+            if (constructor == null)
+            {
+                return CreateInvalidExpressionForHasArgumentsExpression(null, boundObjectCreationExpression.Arguments, boundObjectCreationExpression.InitializerExpressionOpt, syntax, type, constantValue, isImplicit);
+            }
+
             Lazy<IObjectOrCollectionInitializerExpression> initializer = new Lazy<IObjectOrCollectionInitializerExpression>(() => (IObjectOrCollectionInitializerExpression)Create(boundObjectCreationExpression.InitializerExpressionOpt));
             Lazy<ImmutableArray<IArgument>> argumentsInEvaluationOrder = new Lazy<ImmutableArray<IArgument>>(() =>
             {
+                if (boundObjectCreationExpression.BinderOpt == null)
+                {
+                    // binder can only be null for a generated non-null default value of a nullable type parameter, 
```

Comment:
>binder can only be null for a generated non-null default value of a nullable type parameter [](start = 23, length = 91)

Where do we create this node? I think we should make sure it has the binder rather than making an assumption based on the fact that it is missing.

Reply:
I would like to get a better understanding of this scenario, let's talk offline you have time.

---
In reply to: [144118135](https://github.com/dotnet/roslyn/pull/22651#discussion_r144118135) [](ancestors = 144118135)